lib/spack/spack/test/environment_modifications.py

```diff
@@ -158,75 +210,96 @@ def test_unset(env):
-        "/usr/local/opt/some-package/lib",
-        "/usr/opt/lib",
-        "/opt/some-package/include",
-        "/opt/some-package/local/..",
-    ]
-    assert filtered == expected
+    filtered = filter_system_paths(system_paths_for_os + non_system_paths_for_os)
+    assert filtered == non_system_paths_for_os
 
 
-# TODO 27021
-@pytest.mark.not_on_windows("Not supported on Windows (yet)")
-def test_set_path(env):
+@pytest.mark.parametrize("name,elements,separator", [("A", ["foo", "bar", "baz"], os.pathsep)])
+def test_set_path(env, name, elements, separator):
     """Tests setting paths in an environment variable."""
 
-    # Check setting paths with the default separator
-    env.set_path("A", ["foo", "bar", "baz"])
+    # Check setting paths with a specific separator
+    env.set_path(name, elements, separator=separator)
     env.apply_modifications()
 
-    assert "foo:bar:baz" == os.environ["A"]
-
-    env.set_path("B", ["foo", "bar", "baz"], separator=";")
-    env.apply_modifications()
-
-    assert "foo;bar;baz" == os.environ["B"]
+    expected = os.pathsep.join(elements)
+    assert expected == os.environ[name]
 
 
-@pytest.mark.not_on_windows("Not supported on Windows (yet)")
-def test_path_manipulation(env):
+@pytest.mark.parametrize(
+    "path_name,elements,expected",
+    [
+        (
+            "PATH_LIST",
+            ["first", "fourth", "last"],
+            make_pathlist(
+                [
+                    ["path", "first"],
+                    ["path", "second"],
+                    ["path", "third"],
+                    ["path", "fourth"],
+                    ["path", "last"],
+                ]
+            ),
+        ),
+        (
+            "EMPTY_PATH_LIST",
+            ["first", "middle", "last"],
+            make_pathlist([["path", "first"], ["path", "middle"], ["path", "last"]]),
+        ),
+        (
+            "NEWLY_CREATED_PATH_LIST",
+            ["first", "middle", "last"],
+            make_pathlist([["path", "first"], ["path", "middle"], ["path", "last"]]),
+        ),
+    ],
+)
+def test_path_manipulation(env, path_name, elements, expected):
     """Tests manipulating list of paths in the environment."""
 
-    env.append_path("PATH_LIST", "/path/last")
-    env.prepend_path("PATH_LIST", "/path/first")
-
-    env.append_path("EMPTY_PATH_LIST", "/path/middle")
-    env.append_path("EMPTY_PATH_LIST", "/path/last")
-    env.prepend_path("EMPTY_PATH_LIST", "/path/first")
+    env.prepend_path(path_name, make_path("path", elements[0]))
+    env.append_path(path_name, make_path("path", elements[1]))
+    env.append_path(path_name, make_path("path", elements[2]))
 
-    env.append_path("NEWLY_CREATED_PATH_LIST", "/path/middle")
-    env.append_path("NEWLY_CREATED_PATH_LIST", "/path/last")
-    env.prepend_path("NEWLY_CREATED_PATH_LIST", "/path/first")
+    env.remove_path("REMOVE_PATH_LIST", make_path("remove", "this"))
+    env.remove_path("REMOVE_PATH_LIST", make_path("duplicate") + os.sep)
```

Comment:
Can the parameterization be undone for this? Right now this test is a bit confusing because it looks like some parts are re-executed with no variation based on the parameters (technically not a problem, but a bit odd to be doing that).